graph/src/data/graphql/queryable_type.rs

```diff
@@ -0,0 +1,179 @@
+use crate::prelude::s;
+use crate::schema::{EntityType, Schema};
+use std::cmp::Ordering;
+use std::collections::BTreeMap;
+use std::hash::{Hash, Hasher};
+use std::mem;
+
+use super::{DocumentExt, ObjectTypeExt};
+
+#[derive(Copy, Clone, Debug)]
+pub enum QueryableType<'a> {
+    Object(&'a s::ObjectType),
+    Interface(&'a s::InterfaceType),
+    Union(&'a s::UnionType),
+}
+
+impl<'a> PartialEq for QueryableType<'a> {
+    fn eq(&self, other: &Self) -> bool {
+        use QueryableType::*;
+        match (self, other) {
+            (Object(a), Object(b)) => a.name == b.name,
+            (Interface(a), Interface(b)) => a.name == b.name,
+            (Union(a), Union(b)) => a.name == b.name,
+            (Object(_), Union(_))
+            | (Interface(_), Union(_))
+            | (Union(_), Object(_))
+            | (Union(_), Interface(_))
+            | (Interface(_), Object(_))
+            | (Object(_), Interface(_)) => false,
```

Comment:
Rust has exhaustive pattern matching, but it all falls apart when we use the catch-all argument.

We use all the possible enum values so in case we add another one, the compiler will tell us where it's being used.

Reply:
That makes total sense, thanks!